packages/salesforcedx-vscode-lwc/src/commands/forceLightningLwcPreview.ts

```diff
@@ -407,9 +574,50 @@ async function selectPlatformAndExecute(
           previewExecution.command.toString()
         );
         vscode.window.showInformationMessage(
-          nls.localize('force_lightning_lwc_android_start', target)
+          nls.localize('force_lightning_lwc_android_start', targetDevice)
         );
       }
     });
   }
 }
+
+/**
+ * Given a path, it recursively goes through that directory and upwards, until it finds
+ * a config file named sfdx-project.json and returns the path to the folder containg it.
+ *
+ * @param startPath starting path to search for the config file.
+ * @returns the path to the folder containing the config file, or undefined if config file not found
+ */
+function getProjectRootDirectory(startPath: string): string | undefined {
+  const searchingForFile = 'sfdx-project.json';
+
+  let dir: string | undefined = path.parse(startPath).dir;
```

Comment:
This isn't going to be fully accurate if `startPath` itself is a folder—it fails in the case where you're in the project folder already. It works because we're typically going to be buried down a multi-folder path from `sfdx-project.json`, but technically speaking:
```javascript
console.log(`${path.parse('/my/component/folder').dir)}`);
// '/my/component'
```

Reply:
Nice catch